src/Board.java

```diff
@@ -97,4 +106,40 @@ private void updatePieceCount(int position) {
             this.numberOfBlackPieces--;
         }
     }
+    
+    public boolean isEndState() {
+        return (numberOfBlackPieces == 0 || numberOfWhitePieces == 0);
+    }
+
+    public List<Square> getAdjacentSquares(Square square) {
+        List<Integer> squareNumbers = square.getAdjacentSquares();
+        return getSquares((ArrayList<Integer>) squareNumbers);
+    }
+
+    public List<Square> getSquaresTwoMovesAway(Square square) {
+        List<Square> squaresTwoHopsAway = new ArrayList<Square>();
+        for (Square adjacentSquare : getAdjacentSquares(square)) {
+            for (Square twoAway : getAdjacentSquares(adjacentSquare)) {
+                if (!square.isInSameRow(twoAway) && square.isInSameColumn(twoAway)) {
+                    squaresTwoHopsAway.add(twoAway);
+                }
+            }
+        }
+        return squaresTwoHopsAway;
+    }
+
+    public void printBoard() {
+        printSquares(gameState);
+    }
+
+    public void printSquares(List<Square> gameState) {
+        for (int i = 1; i < gameState.size(); i++) {
+            System.out.println(gameState.get(i) + ", " + gameState.get(i).getOccupyingPiece());
+        }
+    }
+
+    public static void main(String[] args) {
```

Comment:
I think this main method should be removed. I suspect this was just for some quick testing.